addOns/brk/src/main/java/org/zaproxy/addon/brk/ExtensionBreak.java

```diff
@@ -0,0 +1,689 @@
+import org.zaproxy.addon.brk.impl.http.HttpBreakpointMessage.Match;
+import org.zaproxy.addon.brk.impl.http.HttpBreakpointsUiManagerInterface;
+import org.zaproxy.addon.brk.impl.http.ProxyListenerBreak;
+import org.zaproxy.zap.extension.help.ExtensionHelp;
+import org.zaproxy.zap.extension.httppanel.Message;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionBreak extends ExtensionAdaptor
+        implements SessionChangedListener, OptionsChangedListener {
+
+    public static final String NAME = "ExtensionBreak";
+
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionBreak.class);
+
+    public static final String BREAK_POINT_HIT_STATS = "stats.break.hit";
+    public static final String BREAK_POINT_STEP_STATS = "stats.break.step";
+    public static final String BREAK_POINT_DROP_STATS = "stats.break.drop";
+
+    private BreakPanel breakPanel = null;
+    private ProxyListenerBreak proxyListener = null;
+
+    private BreakpointsPanel breakpointsPanel = null;
+
+    private PopupMenuEditBreak popupMenuEditBreak = null;
+    private PopupMenuRemove popupMenuRemove = null;
+
+    private BreakpointMessageHandler2 breakpointMessageHandler;
+
+    private Map<Class<? extends BreakpointMessageInterface>, BreakpointsUiManagerInterface>
+            mapBreakpointUiManager;
+
+    private Map<Class<? extends Message>, BreakpointsUiManagerInterface> mapMessageUiManager;
+
+    private Mode mode = Control.getSingleton().getMode();
+
+    private BreakpointsParam breakpointsParams;
+
+    private BreakpointsOptionsPanel breakpointsOptionsPanel;
+
+    private BreakpointManagementInterface breakpointManagementInterface;
+    private HttpBreakpointsUiManagerInterface httpBreakpoints;
+
+    private ZapMenuItem menuBreakOnRequests = null;
+    private ZapMenuItem menuBreakOnResponses = null;
+    private ZapMenuItem menuStep = null;
+    private ZapMenuItem menuContinue = null;
+    private ZapMenuItem menuDrop = null;
+    private ZapMenuItem menuHttpBreakpoint = null;
+
+    private BreakAPI api = new BreakAPI(this);
+    private List<Consumer<Boolean>> serialisationRequiredListeners;
+
+    public ExtensionBreak() {
+        super(NAME);
+        this.setOrder(24);
+    }
+
+    @Override
+    public String getUIName() {
+        return Constant.messages.getString("brk.name");
+    }
+
+    @Override
+    public void init() {
+        serialisationRequiredListeners = Collections.synchronizedList(new ArrayList<>(1));
+        try {
+            ExtensionOption extOption = Control.getSingleton().getExtensionLoader().getExtension(ExtensionOption.class);
+            Method method = extOption.getClass().getDeclaredMethod("setBreakOptionsHandler");
+            method.invoke(extOption, this.getOptionsParam()::setShowIgnoreFilesButtons);
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException(e);
```

Comment:
@thc202 I am not able to understand how to get this method with signature `public void setBreakOptionsHandler(BreakOptionsHandler breakOptionsHandler)` using reflection since I don't have access to BreakOptionsHandler Interface from here.

Reply:
```
> Task :addOns:brk:compileJava FAILED
/home/aryangupta701/Desktop/contributions/zap-extensions/addOns/brk/src/main/java/org/zaproxy/addon/brk/ExtensionBreak.java:125: error: method invoke in class Method cannot be applied to given types;
            method.invoke(extOption, this.getOptionsParam()::setShowIgnoreFilesButtons);
                  ^
  required: Object,Object[]
  found: ExtensionOption,this.getOp[...]ttons
  reason: varargs mismatch; Object is not a functional interface
```

This is the error I am facing

Reply:
You just need to load the class, should be more like:
```java
ExtensionOption extOption =
        Control.getSingleton().getExtensionLoader().getExtension(ExtensionOption.class);
Class<?> breakOptionsHandlerClass =
        getClass()
                .getClassLoader()
                .loadClass("org.parosproxy.paros.extension.option.BreakOptionsHandler");
Method method =
        extOption
                .getClass()
                .getDeclaredMethod("setBreakOptionsHandler", breakOptionsHandlerClass);
method.invoke(
        extOption,
        Proxy.newProxyInstance(
                breakOptionsHandlerClass.getClassLoader(),
                new Class<?>[] {breakOptionsHandlerClass},
                (proxy, m, args) -> {
                    getOptionsParam().setShowIgnoreFilesButtons((Boolean) args[0]);
                    return null;
                }));
```